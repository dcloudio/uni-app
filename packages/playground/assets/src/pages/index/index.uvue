<template>
    <scroll-view direction="vertical">
        <navigator url="/pages/sub/test/test">test</navigator>
        <test-com></test-com>
        <view>
            <view>3:{{ invalidStatic }}</view>
            <image style="width: 50px;height: 50px;" src="./static/pages-index-static_invalid-static.jpg" />
        </view>
        <view>
            <view>4:{{ logo }}</view>
            <image style="width: 50px;height: 50px;" src="../../static/static_logo.png" />
        </view>
    </scroll-view>
</template>
<script>
import invalidStatic from './static/pages-index-static_invalid-static.jpg'
import logo from '../../static/static_logo.png'
export default {
    data() {
        return {
            invalidStatic,
            logo
        }
    }
}
</script>