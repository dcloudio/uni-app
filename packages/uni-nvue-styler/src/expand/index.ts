import type { Plugin } from 'postcss'
import {
  type NormalizeOptions,
  type TransformDecl,
  hyphenateStyleProperty,
} from '../utils'
import { createTransformBackground } from './background'
import { createTransformBorder } from './border'
import { transformBorderColor } from './borderColor'
import { transformBorderRadius } from './borderRadius'
import { transformBorderStyle } from './borderStyle'
import { transformBorderWidth } from './borderWidth'
import { transformFlexFlow } from './flexFlow'
import { transformFont } from './font'
import { transformMargin } from './margin'
import { transformPadding } from './padding'
import { transformTransition } from './transition'

function getDeclTransforms(
  options: NormalizeOptions
): Record<string, TransformDecl> {
  const transformBorder = createTransformBorder(options)
  const styleMap: Record<string, TransformDecl> = {
    transition: transformTransition,
    border: transformBorder,
    background: createTransformBackground(options),
    borderTop: transformBorder,
    borderRight: transformBorder,
    borderBottom: transformBorder,
    borderLeft: transformBorder,
    borderStyle: transformBorderStyle,
    borderWidth: transformBorderWidth,
    borderColor: transformBorderColor,
    borderRadius: transformBorderRadius,
    // uvue已经支持这些简写属性，不需要展开
    // margin,padding继续展开，确保样式的优先级
    margin: transformMargin,
    padding: transformPadding,
    /* eslint-disable no-restricted-syntax */
    ...(options.type !== 'uvue'
      ? {
          flexFlow: transformFlexFlow,
        }
      : {}),
  }
  let result: Record<string, TransformDecl> = {}
  if (__NODE_JS__) {
    styleMap.font = transformFont
    for (const property in styleMap) {
      result[hyphenateStyleProperty(property)] = styleMap[property]
    }
  } else {
    result = styleMap
  }
  return result
}

let DeclTransforms: Record<string, TransformDecl>
const expanded = Symbol('expanded')
export function expand(options: NormalizeOptions): Plugin {
  const plugin: Plugin = {
    postcssPlugin: `${options.type || 'nvue'}:expand`,
    Declaration(decl) {
      if ((decl as any)[expanded]) {
        return
      }
      if (!DeclTransforms) {
        DeclTransforms = getDeclTransforms(options)
      }
      const transform = DeclTransforms[decl.prop]
      if (transform) {
        const res = transform(decl)
        const isSame = res.length === 1 && res[0] === decl
        if (!isSame) {
          decl.replaceWith(res)
        }
      }
      ;(decl as any)[expanded] = true
    },
  }
  return plugin
}
