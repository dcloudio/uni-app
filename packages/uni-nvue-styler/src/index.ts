export { expand } from './expand'
export { normalize } from './normalize'
export { objectifier } from './objectifier'
export { parse } from './parse'
export { parseInlineStyleSync } from './parseSync'
export { checkColor } from './utils'
