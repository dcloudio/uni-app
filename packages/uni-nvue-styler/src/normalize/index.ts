import type { Plugin, Declaration, Helpers, Rule } from 'postcss'
import { camelize, hasOwn, isFunction, isString } from '@vue/shared'
import {
  LENGTH_REGEXP,
  SUPPORT_CSS_UNIT,
  hyphenateStyleProperty,
  isNumber,
  COMBINATORS_RE,
} from '../utils'
import { getNormalizeMap } from './map'

const normalized = Symbol('normalized')

export interface NormalizeOptions {
  logLevel?: 'NOTE' | 'WARNING' | 'ERROR'
  type?: 'nvue' | 'uvue'
}

export function normalize(opts: NormalizeOptions = {}): Plugin {
  if (!hasOwn(opts, 'logLevel')) {
    opts.logLevel = 'WARNING'
  }
  const plugin: Plugin = {
    postcssPlugin: `${opts.type || 'nvue'}:normalize`,
    Declaration: createDeclarationProcessor(opts),
  }
  if (__NODE_JS__) {
    plugin.Rule = createRuleProcessor()
  }
  return plugin
}

function createRuleProcessor() {
  return (rule: Rule, helper: Helpers) => {
    if ((rule as any)[normalized]) {
      return
    }
    rule.selector = rule.selectors
      .map((selector) => {
        selector = selector
          .replace(/\s*([\+\~\>])\s*/g, '$1')
          .replace(/\s+/, ' ')
        if (COMBINATORS_RE.test(selector)) {
          return selector
        }
        rule.warn(
          helper.result,
          'ERROR: Selector `' +
            selector +
            '` is not supported. nvue only support classname selector'
        )
        return ''
      })
      .filter(Boolean)
      .join(', ')
    if (!rule.selector) {
      rule.remove()
    }
    ;(rule as any)[normalized] = true
  }
}

function createDeclarationProcessor(options: NormalizeOptions) {
  return (decl: Declaration, helper: Helpers) => {
    if ((decl as any)[normalized]) {
      return
    }
    decl.prop = camelize(decl.prop)
    const { value, log } = normalizeDecl(decl.prop, decl.value, options)
    if (isString(value) || isNumber(value)) {
      decl.value = value
    }
    if (log && log.reason && helper) {
      const { reason } = log
      let needLog = false
      if (options.logLevel === 'NOTE') {
        needLog = true
      } else if (options.logLevel === 'ERROR') {
        if (reason.startsWith('ERROR:')) {
          needLog = true
        }
      } else {
        if (!reason.startsWith('NOTE:')) {
          needLog = true
        }
      }
      needLog && decl.warn(helper.result, reason)
    }
    if (value === null) {
      decl.remove()
    }
    ;(decl as any)[normalized] = true
  }
}

export function normalizeDecl(
  name: string,
  value: string,
  options: NormalizeOptions
) {
  let result, log
  const normalize = getNormalizeMap(options)[name]

  if (isFunction(normalize)) {
    if (!isFunction(value)) {
      result = normalize(value, options)
    } else {
      result = { value: value }
    }
    if (result.reason) {
      log = { reason: result.reason(name, value, result.value) }
    }
  } else {
    // ensure number type, no `px`
    if (isString(value)) {
      const match = value.match(LENGTH_REGEXP)
      if (match && (!match[1] || SUPPORT_CSS_UNIT.indexOf(match[1]) === -1)) {
        value = parseFloat(value) as unknown as string
      }
    }
    result = { value: value }
    log = {
      reason:
        'WARNING: `' +
        hyphenateStyleProperty(name) +
        '` is not a standard property name (may not be supported)',
    }
  }
  return {
    value: result.value,
    log,
  }
}
