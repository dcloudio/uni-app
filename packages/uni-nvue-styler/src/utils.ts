import type { Declaration, Source } from 'postcss'

export const COMBINATORS_RE =
  /^((?:(?:\.[A-Za-z0-9_\-]+)+[\+\~\> ])*)((?:\.[A-Za-z0-9_\-\:]+)+)$/

export type TransformDecl = (decl: Declaration) => Declaration[]

export interface NormalizeOptions {
  logLevel?: 'NOTE' | 'WARNING' | 'ERROR'
  type?: 'nvue' | 'uvue'
  platform?: typeof process.env.UNI_UTS_PLATFORM
}

export type Normalize = (
  v: string | number,
  options: NormalizeOptions,
  declInfo?: {
    atRule: string
  }
) => {
  value: string | number | null
  reason?: (
    k: string,
    v: string | number,
    result: string | number | null
  ) => string
}

export const enum Restriction {
  ANGLE = 'angle', // "角度"
  BOX = 'box', // "区域"
  COLOR = 'color', // "颜色"
  ENUM = 'enum', // "枚举"
  FONT = 'font', // "字体"
  GEOMETRY_BOX = 'geometry-box', // "几何框"
  IDENTIFIER = 'identifier', // "标识符"
  IMAGE = 'image', // "图片"
  INTEGER = 'integer', // "整数"
  LENGTH = 'length', // "长度"
  LINE_STYLE = 'line-style', // "样式"
  LINE_WIDTH = 'line-width', // "长度"
  NUMBER = 'number', // "数字"
  NUMBER_0_1 = 'number(0-1)', // "数字"
  PERCENTAGE = 'percentage', // "百分比"
  POSITION = 'position', // "位置"
  PROPERTY = 'property', // "属性"
  REPEAT = 'repeat', // "重复"
  SHAPE = 'shape', // "区域大小"
  STRING = 'string', // "字符串"
  TIME = 'time', // "时间"
  TIMING_FUNCTION = 'timing-function', // "时间函数"
  UNICODE_RANGE = 'unicode-range', // "编码范围"
  URL = 'url', // "超链接"
  GRADIENT = 'gradient', // "渐变"
}

export interface UniPlatform {
  app: {
    android?: {
      osVer?: string
      uniVer?: string
      unixVer?: string
    }
    ios?: {
      osVer?: string
      uniVer?: string
      unixVer?: string
    }
    harmony?: {
      osVer?: string
      uniVer?: string
      unixVer?: string
    }
  }
}

export interface PropertyValue {
  name: string
  uniPlatform?: UniPlatform
}

export interface Property {
  name: string
  shorthand?: boolean
  restrictions: Restriction[]
  values?: PropertyValue[]
  uniPlatform?: UniPlatform
  unixTags?: string[]
}

export interface CssJSON {
  version: number
  properties: Property[]
}

export function createDecl(
  prop: string,
  value: string,
  important: boolean,
  raws: Record<string, unknown>,
  source?: Source
) {
  const decl = {
    type: 'decl',
    prop,
    value: value.toString(),
    raws,
    source,
  } as Declaration
  if (important) {
    decl.important = true
  }
  return decl
}

export const NUM_REGEXP = /^[-]?\d*\.?\d+$/
export const LENGTH_REGEXP = /^[-+]?\d*\.?\d+(\S*)$/
export const SUPPORTED_VALUES_REGEXP = /supported values are: ([^)]+)/
export const SUPPORT_CSS_UNIT = ['px', 'pt', 'wx', 'upx', 'rpx']

export const isNumber = (val: unknown): val is string => typeof val === 'number'

export const serialize = (val: unknown) => JSON.parse(JSON.stringify(val))

const cacheStringFunction = <T extends (str: string) => string>(fn: T): T => {
  const cache: Record<string, string> = Object.create(null)
  return ((str: string) => {
    const hit = cache[str]
    return hit || (cache[str] = fn(str))
  }) as any
}

const hyphenateRE = /([A-Z])/g

export const hyphenateStyleProperty = cacheStringFunction((str: string) =>
  str
    .replace(hyphenateRE, (_, m) => {
      if (typeof m === 'string') {
        return '-' + m.toLowerCase()
      }
      return m
    })
    .toLowerCase()
)

export function autofixedReason(
  v: string | number,
  result: string | number | null
) {
  return 'NOTE: property value `' + v + '` is autofixed to `' + result + '`'
}

export function validReason(k: string, v: string | number) {
  return (
    'ERROR: property value `' +
    v +
    '` is not valid for `' +
    hyphenateStyleProperty(k) +
    '`'
  )
}

export function defaultValueReason(k: string, v: string | number) {
  return (
    'NOTE: property value `' +
    v +
    '` is the DEFAULT value for `' +
    hyphenateStyleProperty(k) +
    '` (could be removed)'
  )
}

export function supportedEnumReason(
  k: string,
  v: string | number,
  items: unknown[]
) {
  return (
    'ERROR: property value `' +
    v +
    '` is not supported for `' +
    hyphenateStyleProperty(k) +
    '` (supported values are: `' +
    items.join('`|`') +
    '`)'
  )
}

export function supportedValueWithTipsReason(
  k: string,
  v: string | number,
  tips?: string
) {
  return (
    'ERROR: property value `' +
    v +
    '` is not supported for `' +
    hyphenateStyleProperty(k) +
    '` ' +
    tips
  )
}

export function supportedUnitWithAutofixedReason(
  unit: string,
  v: string | number,
  result: string | number | null
) {
  return (
    'NOTE: unit `' +
    unit +
    '` is not supported and property value `' +
    v +
    '` is autofixed to `' +
    result +
    '`'
  )
}

export function compatibilityReason(k: string) {
  return (
    'NOTE: the ' +
    hyphenateStyleProperty(k) +
    ' property may have compatibility problem on native'
  )
}

export function supportedPropertyReason(k: string) {
  return (
    'WARNING: `' +
    hyphenateStyleProperty(k) +
    '` is not a standard property name (may not be supported)'
  )
}

export function getSupportedPlatforms(uniPlatform: UniPlatform | undefined) {
  const supportedPlatforms: string[] = []
  if (uniPlatform?.app?.android?.unixVer !== 'x') {
    supportedPlatforms.push('app-android')
  }
  if (uniPlatform?.app.ios?.unixVer !== 'x') {
    supportedPlatforms.push('app-ios')
  }
  if (uniPlatform?.app.harmony?.unixVer !== 'x') {
    supportedPlatforms.push('app-harmony')
  }
  return supportedPlatforms
}

export function normalizeReasons(
  reasons: string[],
  k: string,
  v: string | number
) {
  let enums: string[] = []
  for (let i = 0; i < reasons.length; i++) {
    const reason = reasons[i]
    if (SUPPORTED_VALUES_REGEXP.test(reason)) {
      const match = reason.match(SUPPORTED_VALUES_REGEXP)
      if (match) {
        const values = match[1]
          .split('|')
          .map((item) => item.replace(/`/g, ''))
          .filter(Boolean)
        enums.push(...values)
        reasons.splice(i, 1)
        i--
      }
    }
  }
  if (enums.length > 0) {
    enums = [...new Set(enums)]
    reasons.push(supportedEnumReason(k, v, enums))
  }
  return reasons
}
