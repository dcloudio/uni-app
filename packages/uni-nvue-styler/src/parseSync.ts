import type { Declaration } from 'postcss'
import { expand } from './expand'
import { normalize } from './normalize'
import type { NormalizeOptions } from './utils'

export interface ParseInlineStyleOptions extends NormalizeOptions {}

export function parseInlineStyleSync(
  input: string,
  options: ParseInlineStyleOptions = {}
) {
  const styleObj = parseStringStyle(input)
  const declarations: Declaration[] = []
  Object.entries(styleObj).forEach(([key, value]) => {
    declarations.push({
      prop: key,
      value: value,
    } as Declaration)
  })
  const expandedDeclarations: Declaration[] = []
  const { Declaration: expandDeclaration } = expand(options)
  if (typeof expandDeclaration === 'function') {
    declarations.forEach((declaration) => {
      expandedDeclarations.push(
        ...visit(declaration, expandDeclaration as (decl: Declaration) => void)
      )
    })
  } else {
    expandedDeclarations.push(...declarations)
  }
  const normalizedDeclarations: Declaration[] = []
  const { Declaration: normalizeDeclaration } = normalize(options)
  if (typeof normalizeDeclaration === 'function') {
    expandedDeclarations.forEach((declaration) => {
      normalizedDeclarations.push(
        ...visit(
          declaration,
          normalizeDeclaration as (decl: Declaration) => void
        )
      )
    })
  } else {
    normalizedDeclarations.push(...expandedDeclarations)
  }
  const styleEntries = normalizedDeclarations.map(
    ({ prop, value }) => `['${prop}', ${JSON.stringify(value)}]`
  )
  return `new Map([${styleEntries.join(', ')}])`
}

function visit(
  declaration: Declaration,
  transformDecl: (decl: Declaration) => void
) {
  let removed = false
  let replaced = false
  const result: Declaration[] = []
  declaration.replaceWith = function (nodes) {
    replaced = true
    result.push(...(nodes as Declaration[]))
    return this
  }
  declaration.remove = function () {
    removed = true
    return this
  }
  transformDecl(declaration)
  if (!removed && !replaced) {
    result.push(declaration)
  }
  return result
}

type NormalizedStyle = Record<string, string | number>

const listDelimiterRE = /;(?![^(]*\))/g
const propertyDelimiterRE = /:([^]+)/
const styleCommentRE = /\/\*[^]*?\*\//g

export function parseStringStyle(cssText: string): NormalizedStyle {
  const ret: NormalizedStyle = {}
  cssText
    .replace(styleCommentRE, '')
    .split(listDelimiterRE)
    .forEach((item) => {
      if (item) {
        const tmp = item.split(propertyDelimiterRE)
        tmp.length > 1 && (ret[tmp[0].trim()] = tmp[1].trim())
      }
    })
  return ret
}
