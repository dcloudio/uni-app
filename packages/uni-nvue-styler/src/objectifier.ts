import type { Container, Document, Root } from 'postcss'
import { extend, hasOwn } from '@vue/shared'
import { COMBINATORS_RE } from './utils'

interface ObjectifierContext {
  'FONT-FACE': Record<string, unknown>[]
  TRANSITION: Record<string, Record<string, unknown>>
}

export function objectifier(
  node: Root | Document | Container | null,
  { trim }: { trim: boolean } = { trim: false }
) {
  if (!node) {
    return {}
  }
  const context: ObjectifierContext = {
    'FONT-FACE': [],
    TRANSITION: {},
  }
  const result = transform(node, context)
  if (trim) {
    trimObj(result)
  }
  if (context['FONT-FACE'].length) {
    result['@FONT-FACE'] = context['FONT-FACE']
  }
  if (Object.keys(context.TRANSITION).length) {
    result['@TRANSITION'] = context.TRANSITION
  }
  return result
}

function trimObj(obj: Record<string, any>) {
  Object.keys(obj).forEach((name) => {
    const value = obj[name]
    if (Object.keys(value).length === 1 && hasOwn(value, '')) {
      obj[name] = value['']
    }
  })
}

function transform(
  node: Root | Document | Container,
  context: ObjectifierContext
) {
  const result: Record<string, Record<string, unknown> | unknown> = {}
  node.each((child) => {
    if (child.type === 'atrule') {
      const body = transform(child, context)
      if (child.name === 'font-face') {
        const fontFamily = body.fontFamily as string
        if (fontFamily && '"\''.indexOf(fontFamily[0]) > -1) {
          body.fontFamily = fontFamily.slice(1, fontFamily.length - 1)
        }
        context['FONT-FACE'].push(body)
      }
    } else if (child.type === 'rule') {
      const body = transform(child, context)
      child.selectors.forEach((selector) => {
        transformSelector(selector, body, result, context)
      })
    } else if (child.type === 'decl') {
      if (child.important) {
        result['!' + child.prop] = child.value
        // !important的值域优先级高，故删除非!important的值域
        delete result[child.prop]
      } else {
        if (!hasOwn(result, '!' + child.prop)) {
          result[child.prop] = child.value
        }
      }
    }
  })
  return result
}

function transformSelector(
  selector: string,
  body: Record<string, unknown>,
  result: Record<string, unknown | Record<string, unknown>>,
  context: ObjectifierContext
) {
  const res = selector.match(COMBINATORS_RE)
  if (!res) {
    return
  }
  let parentSelector = res[1]
  let curSelector = res[2].substring(1)
  // .a.b => a.b
  const dotIndex = curSelector.indexOf('.')
  if (dotIndex > -1) {
    parentSelector += curSelector.substring(dotIndex)
    curSelector = curSelector.substring(0, dotIndex)
  }

  const pseudoIndex = curSelector.indexOf(':')
  if (pseudoIndex > -1) {
    const pseudoClass = curSelector.slice(pseudoIndex)
    curSelector = curSelector.slice(0, pseudoIndex)
    Object.keys(body).forEach(function (name) {
      body[name + pseudoClass] = body[name]
      delete body[name]
    })
  }
  transition(curSelector, body, context)
  if (!Object.keys(body).length) {
    return
  }
  result = (result[curSelector] || (result[curSelector] = {})) as Record<
    string,
    unknown
  >

  if (result[parentSelector]) {
    // clone
    result[parentSelector] = processImportant(
      extend({}, result[parentSelector], body)
    )
  } else {
    result[parentSelector] = body
  }
}

/**
 * 处理 important 属性，如果某个属性是 important，需要将非 important 的该属性移除掉
 * @param body
 */
function processImportant(body: Record<string, unknown>) {
  Object.keys(body).forEach((name) => {
    if (name.startsWith('!')) {
      delete body[name.substring(1)]
    }
  })
  return body
}

function transition(
  className: string,
  body: Record<string, unknown>,
  { TRANSITION }: ObjectifierContext
) {
  Object.keys(body).forEach((prop) => {
    if (prop.indexOf('transition') === 0 && prop !== 'transition') {
      const realProp = prop.replace('transition', '')
      TRANSITION[className] = TRANSITION[className] || {}
      TRANSITION[className][realProp[0].toLowerCase() + realProp.slice(1)] =
        body[prop]
    }
  })
}
