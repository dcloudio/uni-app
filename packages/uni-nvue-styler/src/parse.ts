import postcss, { Message } from 'postcss'
import { objectifier } from './objectifier'
import { expand } from './expand'
import { NormalizeOptions, normalize } from './normalize'

interface ParseOptions extends NormalizeOptions {
  filename?: string
  map?: boolean
  ts?: boolean
  noCode?: boolean
}

export async function parse(input: string, options: ParseOptions = {}) {
  const { root, messages } = await postcss([expand, normalize(options)])
    .process(input, {
      from: options.filename || 'foo.css',
    })
    .catch((err: any) => {
      return {
        root: null,
        messages: [
          {
            type: 'error',
            text: err.message,
          } as Message,
        ],
      }
    })
  if (options.noCode === true) {
    return { code: '', messages }
  }
  const obj = root ? objectifier(root) : {}
  if (options.map) {
    return { code: mapToInitString(objToMap(obj), options.ts, true), messages }
  }
  return { code: JSON.stringify(obj), messages }
}

function mapToInitString(
  map: Map<string, unknown>,
  ts: boolean = false,
  isRoot: boolean = false
): string {
  let entries = []
  for (let [key, value] of map) {
    if (value instanceof Map) {
      entries.push(`["${key}", ${mapToInitString(value, ts)}]`)
    } else {
      entries.push(`["${key}", ${JSON.stringify(value)}]`)
    }
  }
  return `new Map${
    ts
      ? isRoot
        ? '<string, Map<string, Map<string, any>>>'
        : '<string, any>'
      : ''
  }([${entries.join(', ')}])`
}

function objToMap(obj: Record<string, unknown>) {
  const map = new Map()
  for (const key in obj) {
    const value = obj[key]
    if (typeof value === 'object') {
      map.set(key, objToMap(value as Record<string, unknown>))
    } else {
      map.set(key, value)
    }
  }
  return map
}
