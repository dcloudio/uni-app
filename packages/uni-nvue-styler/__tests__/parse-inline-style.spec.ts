import { parseInlineStyleSync } from '../src/index'

describe('inline style', () => {
  test('app-harmony', () => {
    const code = parseInlineStyleSync(
      `
        background-color: red;border:1px solid red;
      `,
      { type: 'uvue', platform: 'app-harmony' }
    )
    expect(code).toMatchSnapshot()
  })
})
