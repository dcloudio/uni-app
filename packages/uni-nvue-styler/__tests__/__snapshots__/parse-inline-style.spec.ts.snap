// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inline style app-harmony 1`] = `"new Map([['backgroundColor', "#FF0000"], ['borderTopWidth', 1], ['borderRightWidth', 1], ['borderBottomWidth', 1], ['borderLeftWidth', 1], ['borderTopStyle', "solid"], ['borderRightStyle', "solid"], ['borderBottomStyle', "solid"], ['borderLeftStyle', "solid"], ['borderTopColor', "#FF0000"], ['borderRightColor', "#FF0000"], ['borderBottomColor', "#FF0000"], ['borderLeftColor', "#FF0000"]])"`;
